src/services/pool/pool.repository.ts

```diff
@@ -57,4 +57,16 @@ export default class PoolRepository {
       },
     };
   }
+
+  private buildProviders() {
+    const providers: SDKPoolRepository[] = [];
+    if (isBalancerApiDefined()) {
+      const balancerApiRepository = this.initializeDecoratedAPIRepository();
+      providers.push(balancerApiRepository);
+    }
+    const subgraphRepository = this.initializeDecoratedSubgraphRepository();
+    providers.push(subgraphRepository);
+
+    return providers;
+  }
```

Comment:
Can we try to define this function in one place?

Reply:
@garethfuller It is not trivial so I suggest that @timjrobinson or me try to do it later in a separate PR.

Reply:
Can usePoolsQuery not use `PoolRepository`? Is it doing something different?

Reply:
The current code shows that `initializeDecoratedAPIRepository` and `initializeDecoratedSubgraphRepository` are slightly different. Probably refactoring to use PoolRepository in both places would avoid that duplication but I'm not so familiar with this code and that's why I check it carefully check it in a different PR.

Reply:
Yea I'd rather just get this merged and do that refactor later.

Reply:
@garethfuller If you agree, could you approve this PR so that we can merge?